pkg/reconcile.go

```diff
@@ -37,17 +37,94 @@ func newRunnerConfig() *runnerConfig {
 
 // Integration describes the set of methods Integrations must implement
 type Integration interface {
-	Diff() ([]Diff, error)
-	Reconcile() error
-	Setup() error
+	CurrentState(context.Context, *ResourceInventory) error
+	DesiredState(context.Context, *ResourceInventory) error
+	Reconcile(context.Context, *ResourceInventory) error
+	Setup(context.Context) error
+}
+
+type Action string
```

Comment:
Is this line needed? Tried looking through to see if it's being referenced anywhere but I can't seem to find it

Reply:
Good spot! deleted